source/_posts/2018-01-03-using-git-hooks-to-improve-your-day-to-day-workflow.markdown

```diff
@@ -0,0 +1,108 @@
+---
+layout: post
+title: "Using git hooks to improve your day-to-day workflow"
+category: git
+comments: true
+author:
+  name: Federico Kauffman
+  email: federico@wyeworks.com
+  twitter_handle: fedekauffman
+  github_handle: fedekau
+  image: /images/team/federico-kauffman.jpg
+  description: Software Engineer at Wyeworks. Currently working with Javascript and Ruby. Learnaholic.
+---
+
+If you have been developing software for some time you have probably noticed that there are lot of things that can go wrong, no matter how hard you try there is always something you might forget because after all, we are just humans doing an extremly difficult task: Telling a computer what to do.
+
+In this post I will expose some of the **git-hooks** we use in some projects here at Wyeworks to make developers life easier by preventing bad commits to even leave their computers. I will cover *linting*, *tests* and *commit formating* use cases.
+
+<!--more-->
+
+## What are git-hooks?
+
+Like many other Version Control Systems, Git has a way to fire off custom scripts when certain important actions occur. There are two groups of these hooks: *client-side* and *server-side*. Client-side hooks are triggered by operations such as committing and merging, while server-side hooks run on network operations such as receiving pushed commits. You can use these hooks for all sorts of reasons.
+
+## Installing a Hook
+
+The hooks are all stored in the hooks subdirectory of the Git directory. In most projects, that’s `.git/hooks`. When you initialize a new repository with `git init`, Git populates the hooks directory with a bunch of example scripts, many of which are useful by themselves; but they also document the input values of each script. All the examples are written as shell scripts, with some Perl thrown in, but any properly named executable scripts will work fine – you can write them in Ruby or Python or whatever language you are familiar with. If you want to use the bundled hook scripts, you’ll have to rename them; their file names all end with `.sample`.
+
+To enable a hook script, put a file in the hooks subdirectory of your `.git` directory that is named appropriately (without any extension) and is executable. From that point forward, it should be called.
+
+In this post we will only be using some *client-side* hooks to execute custom scripts. Next are the hooks and use cases I like the most:
+
+### PRE-COMMIT
+
+The `pre-commit` hook is run first, before you even type in a commit message. It’s used to inspect the snapshot that’s about to be committed, to see if you’ve forgotten something, to make sure tests run, or to examine whatever you need to inspect in the code. Exiting non-zero from this hook aborts the commit, although you can bypass it with `git commit --no-verify`. You can do things like check for code style (run lint or something equivalent), check for trailing whitespace, or check for appropriate documentation on new methods.
```

Comment:
This paragraph and others were copied/pasted from https://git-scm.com/book/gr/v2/Customizing-Git-Git-Hooks. If we don't want to rephrase the hook definitions, we must quote (using the markdown ">") and give credit to the git documentation.